addon/components/form-section.js

```diff
@@ -0,0 +1,159 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { A } from '@ember/array';
+import { getChildrenForSection } from '../utils/model-factory';
+import { guidFor } from '@ember/object/internals';
+import { validationResultsForField } from '@lblod/submission-form-helpers';
+import { next } from '@ember/runloop';
+import isLast from '@lblod/ember-submission-form-fields/-private/helpers/is-last';
+import {
+  PROPERTY_GROUP_DISPLAY_TYPE,
+  SECTION_DISPLAY_TYPE,
+} from '../models/section';
+import { LISTING_TYPE } from '../models/listing';
+
+export default class SubmissionFormSectionComponent extends Component {
+  @tracked children = A();
+  @tracked validations = [];
+  isLast = isLast;
+
+  observerLabel = `section-${guidFor(this)}`;
+
+  constructor() {
+    super(...arguments);
+
+    next(this, () => {
+      this.update(this.section, {
+        form: this.args.form,
+        store: this.args.formStore,
+        graphs: this.args.graphs,
+        node: this.args.sourceNode,
+      });
+    });
+  }
+
+  get section() {
+    return this.args.section || this.args.group;
+  }
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    this.deregister();
+  }
+
+  get level() {
+    return this.section.options.level || this.args.level || 1;
+  }
+
+  get titleLevel() {
+    return `${this.level}`;
+  }
+
+  get titleSkin() {
+    return this.section.options.skin || `${this.level + 1}`;
+  }
+
+  get nextLevel() {
+    return this.level + 1;
+  }
+
+  get showTitleBlock() {
+    return Boolean(this.section.name) || Boolean(this.section.help);
+  }
+
+  get errors() {
+    return this.validations.filter((r) => !r.valid);
+  }
+
+  get storeOptions() {
+    return {
+      formGraph: this.args.graphs.formGraph,
+      sourceGraph: this.args.graphs.sourceGraph,
+      metaGraph: this.args.graphs.metaGraph,
+      sourceNode: this.args.sourceNode,
+      store: this.args.formStore,
+      path: this.section.rdflibPath,
+    };
+  }
+
+  childIsSection(child) {
+    return (
+      child &&
+      [SECTION_DISPLAY_TYPE, PROPERTY_GROUP_DISPLAY_TYPE].includes(
+        child.displayType
+      )
+    );
+  }
+
+  childIsListing(child) {
+    return child && child.displayType === LISTING_TYPE;
+  }
```

Comment:
We still officially support Ember 3.28 in this addon, so this won't work unless we also install the https://github.com/ember-polyfills/ember-functions-as-helper-polyfill as a dependency.

I would just use the helper wrapper instead for now.